From 5879fbd6b97c88d547cf8200bce5696bf8579bc0 Mon Sep 17 00:00:00 2001
From: mnlcz <mnlcz@tuta.io>
Date: Sat, 9 May 2026 15:42:01 -0300
Subject: [PATCH] Expose focuse win title to /tmp/dwl-title

---
 dwl.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/dwl.c b/dwl.c
index 797a66e..e30e43b 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1442,6 +1442,12 @@ focusclient(Client *c, int lift)
 	client_activate_surface(client_surface(c), 1);
 	if (c->foreign_toplevel)
 		wlr_foreign_toplevel_handle_v1_set_activated(c->foreign_toplevel, 1);
+	const char *title = client_get_title(c);
+	FILE *tf = fopen("/tmp/dwl-title", "w");
+	if (tf) {
+		fprintf(tf, "%s", title ? title : "");
+		fclose(tf);
+	}
 }
 
 void
@@ -2983,8 +2989,15 @@ updatetitle(struct wl_listener *listener, void *data)
 			title = broken;
 		wlr_foreign_toplevel_handle_v1_set_title(c->foreign_toplevel, title);
 	}
-	if (c == focustop(c->mon))
+	if (c == focustop(c->mon)) {
 		printstatus();
+		const char *title = client_get_title(c);
+		FILE *tf = fopen("/tmp/dwl-title", "w");
+		if (tf) {
+			fprintf(tf, "%s", title ? title : "");
+			fclose(tf);
+		}
+	}
 }
 
 void
-- 
2.47.3

