From 5acd19007107aeeed48abe365b0e3e5799d01ff9 Mon Sep 17 00:00:00 2001
From: mnlcz <mnlcz@tuta.io>
Date: Wed, 29 Apr 2026 08:28:47 -0300
Subject: [PATCH] Only redraw status bar when content changes

---
 dwlb.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/dwlb.c b/dwlb.c
index faab2d9..5eadf3a 100644
--- a/dwlb.c
+++ b/dwlb.c
@@ -1497,15 +1497,20 @@ read_socket(void)
 			wl_list_for_each(bar, &bar_list, link) {
 				if (first) {
 					copy_customtext(&first->status, &bar->status);
+					bar->redraw = true;
 				} else {
-					parse_into_customtext(&bar->status, wordend);
-					first = bar;
+					if (strcmp(bar->status.text, wordend) != 0) {
+						parse_into_customtext(&bar->status, wordend);
+						first = bar;
+						bar->redraw = true;
+					}
 				}
-				bar->redraw = true;
 			}
 		} else {
-			parse_into_customtext(&bar->status, wordend);
-			bar->redraw = true;
+			if (strcmp(bar->status.text, wordend) != 0) {
+				parse_into_customtext(&bar->status, wordend);
+				bar->redraw = true;
+			}
 		}
 	} else if (!strcmp(wordbeg, "title")) {
 		if (!custom_title || !*wordend)
-- 
2.47.3

